.github/workflows/ci-new.yml

```diff
@@ -248,3 +273,27 @@ jobs:
             -shuffle=on \
             -timeout='10m' \
             ./test/integ-new/...
+
+  cleanup:
```

Comment:
What if integration failed in the beginning and the key is not yet created. I would suggest just add a step in integration job at the end and the condition would be `if: '${{ always() && steps.create_kms_key.outcome == ''success'' }}'`, note you need to set a step id `id: create_kms_key` in your create_kms_key step.